crates/protocol/build.rs

```diff
@@ -20,6 +20,31 @@ struct TypeAttrs<'a> {
 /// https://docs.rs/prost-build/0.6.1/prost_build/struct.Config.html#arguments
 /// `field_attrs` holds tuples of field name to field attributes.
 static TYPE_ATTRS: &'static [TypeAttrs<'static>] = &[
+    TypeAttrs {
```

Comment:
_do_ these need to serde ? aren't we passing them around using protobuf encoding from the Rust side ?

Reply:
Whoops, I meant to add a comment about that. These only need to serde for the snapshot tests, since the `debug_snapshot` variants don't support redaction. I'll propose adding a comment here and also changing these structs to use `cfg_attr(test, derive(Serialize, Deserialize))` so that it's more clear.

Reply:
👍

Reply:
Just to close the loop on this, using `cfg_addr(test, ...)` doesn't quite work, because the tests are in the `build` crate, and Cargo doesn't compile a crate's dependencies using the `test` config, only the crate that's actually under test. So I did this by adding a `test-support` feature to `protocol`, which is enabled as part of `dev-dependencies` in the `build` crate.